### PR TITLE
Add mpackkey decorator

### PR DESCRIPTION
**Pull request by mlsteele**

### CODE REVIEW

File: src/emit.iced [hunk 1]
```diff
@@ -113,18 +113,20 @@ exports.GoEmitter = class GoEmitter
...
     true
 
-  codec : ({name, optional, jsonkey}) ->
+  codec : ({name, optional, jsonkey, mpackkey}) ->
     omitempty = if optional then ",omitempty" else ""
     unless jsonkey?
       jsonkey = "#{name}"
-    "`codec:\"#{name}#{omitempty}\" json:\"#{jsonkey}#{omitempty}\"`"
+    unless mpackkey?
+      mpackkey = "#{name}"
```
Comment (maxtaco) [inline src/emit.iced]: why not just `mpackkey = name`?

Comment (mlsteele) [reply src/emit.iced]: because if it's set we want the value of mpackkey

Comment (maxtaco) [reply src/emit.iced]: sorry, i meant why not:

```iced
unless mpackkey?
  mpackkey = name
```

Comment (mlsteele) [reply src/emit.iced]: ohh idk I'll change them both

**Pull request merged**